client-js/app/elements/labrad-plot.ts

```diff
@@ -654,56 +685,185 @@ export class Plot extends polymer.Base {
 
 
   /**
-   * Generate the geometries and materials for a 1D plot.
+   * Helper method to get the scene object, or create it if it doesn't already
+   * exist.
    */
-  private plotData1D(data: number[][]): void {
-    this.dataLimits1D(data);
-
-    if (this.sceneObjects.length == 0) {
+  private getSceneObject() {
+    if (this.sceneObjects.length === 0) {
       const ob = new THREE.Object3D();
       this.sceneObjects.push(ob);
       this.scene.add(ob);
     }
 
-    const ob = this.sceneObjects[0];
+    return this.sceneObjects[0];
+  }
+
+
+  /**
+   * Plots a line given a set of data, adding it to the scene.
+   */
+  private plotLine(data: number[][], yColumn: number = 1,
+                   color: string = "#000", lineWidth: number = 1,
+                   lines: THREE.Line[]) {
```

Comment:
I would suggest just returning the line and letting the caller push it onto the appropriate list. Also note that it's best not to put required arguments after arguments with default values.